### ISSUE TITLE
getAuthorizationCode error when trying to migrate from MWS token

### ISSUE THREAD

**Issue by chrismeats** [jlevers/selling-partner-api#110]
I referenced this in issue #57 but was not sure you would see it since that issue is closed

@jlevers I am still getting these same error messages on the newest release 4.0.2
if I leave lwaRefreshToken blank I get
{"error_description":"The request is missing a required parameter : refresh_token","error":"invalid_request"}
If I remove lwaRefreshToken from my Configuration then I get
lwaRefreshToken must be specified when calling non-grantless API operations
Here is my config I am using:

$config = new Configuration([
            "lwaClientId"        => env('LWA_CLIENT_ID'),
            "lwaClientSecret"    => env('LWA_CLIENT_SECRET'),
//            "lwaRefreshToken"    => '',
            "awsAccessKeyId"     => env('AMAZON_ACCESS_KEY_ID'),
            "awsSecretAccessKey" => env('AMAZON_SECRET_ACCESS_KEY'),
            "endpoint"           => Endpoint::NA,  // or another endpoint from lib/Endpoints.php
            "roleArn"            => env('SPAPI_ROLE_ARN'),
            "accessToken"        => null,
        ]);
What am I missing here?
Any help is GREATLY appreciated!

**Comment by gokigoks** [comment]
I am also having trouble with the getAuthorizationCode endpoint.. In the link @chrismeats posted of the older issue, I put a random value on the refresh token parameter; however, I am getting this instead:
```

Client error: `POST https://api.amazon.com/auth/o2/token` resulted in a `400 Bad Request` response:
{"error_description":"The request has an invalid grant parameter : refresh_token","error":"invalid_grant"}
```

**Comment by jlevers** [comment]
Give `v4.0.3` a try -- this problem should be fixed. Let me know if you run into any more issues!

**Comment by chrismeats** [comment]
hey @jlevers I hate to bother you again, but I think there is still an issue with `getAuthorizationCode`.
I am getting the following error now
```[403] {
  "errors": [
    {
      "message": "Access to requested resource is denied.",
     "code": "Unauthorized",
     "details": ""
    }
  ]
}
```
I was getting this before and assumed it was because my app had not been approved by amazon yet. They approved my app yesterday and the status now shows 
```
Published
Current edit is approved and pending publishing
```

I can make calls to other spapi authorized endpoints such as getOrders.

Is it possible that the get authorization request has an issue with how it is signed?
Or is my issue that it is showing "Current edit is approved and pending publishing" even though the status shows "published"

Thanks again for an amazing package, let me know if you would rather me start a new issue to track this?

Thanks,

Chris

**Comment by jlevers** [comment]
I think that once your edits have been approved, the `getAuthorizationCode` call should start working again.

(And no worries, this issue is fine -- I'm pretty sure this is an Amazon issue and not one related to my library, so having it in a closed issue actually works well.)

**Comment by gokigoks** [comment]
@jlevers 

Our app has been in 

"Published
Current edit is in draft"

state for a long time. What should the status show to indicate we are good to go with SP-API? Our App is a hybrid app by the way. Any help is appreciated..

**Comment by misterakko** [comment]
@gokigoks I've seen "Current edit is in draft" for as much as two months at a time. Once it got stuck indefinitely, and we had to open a case to unblock it.
I may be wrong, but I suspect there's some poor soul somewhere in India who has been tasked to read and approve every change made, to ensure that nobody added curses, insults at mr. Bezos or black magic to their entries.

**Comment by jlevers** [comment]
@gokigoks as @misterakko said, it's basically going to stay that way until they finally get around to dealing with it *or* you make a stink (creating a support case, making an issue in their GitHub repo, etc).

**Comment by gokigoks** [comment]
@ramsesIII 

So far, Amazon error responses have been very unhelpful most of the time. Try dumping the parameters you are sending in your request for that endpoint and check if it looks good. Good luck.

**Comment by ramsesIII** [comment]
But the absurd thing is that if I put EU_SANDBOX it returns me a token. (not valid but correct payload)
If I return to the EU, it gives me an error 500

**Comment by jlevers** [comment]
I recommend opening a case with Amazon seller support -- a 500 error is unfortunately not something I can do much about. That's an error on their end.

**Comment by ruskenberry** [comment]
Anyway to allow the getAuthorization() call to point to the version=beta url? I see that as a possible workaround for private sellers to manage multiple seller accounts from one MWS developer profile here. 
https://github.com/amzn/selling-partner-api-docs/issues/651#issuecomment-826004377

**Comment by jlevers** [comment]
Not that I know of. If you want to manage 10 or fewer seller accounts, and you have the logins for all of them, you can self-authorize them all. If you have 25 or fewer, you can authorize them using the OAuth flow with the `version=beta` parameter, but you can't convert MWS tokens to SP API tokens.
